autofs-5.0.6 - fix fix map source check in file lookup

From: Ian Kent <raven@themaw.net>

A recent change to correct a problem with included map entry removal
has broken a different case of included map key lookup. The check in
previous patch was too broad and caused map key lookup for keys in an
included multi-mount map entrys to not be found.
---

 CHANGELOG             |    1 +
 modules/lookup_file.c |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)


--- autofs-5.0.5.orig/CHANGELOG
+++ autofs-5.0.5/CHANGELOG
@@ -76,6 +76,7 @@
 - fix ipv6 configure check.
 - add sss lookup module.
 - teach automount about sss source.
+- fix fix map source check in file lookup.
 
 03/09/2009 autofs-5.0.5
 -----------------------
--- autofs-5.0.5.orig/modules/lookup_file.c
+++ autofs-5.0.5/modules/lookup_file.c
@@ -1048,7 +1048,7 @@ do_cache_lookup:
 	 * instance (same map entry cache), not in a distinct source.
 	 */
 	if (me && (!me->mapent || 
-	   (ap->type == LKP_INDIRECT && me->source != source))) {
+	   (me->source != source && *me->key != '/'))) {
 		while ((me = cache_lookup_key_next(me)))
 			if (me->source == source)
 				break;
